src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerAgreementController.cs

```diff
@@ -128,6 +129,13 @@ public async Task<ActionResult> SignAgreement(GetEmployerAgreementRequest reques
             var response = await _mediator.SendAsync(request);
             var viewModel = _mapper.Map<GetEmployerAgreementResponse, EmployerAgreementViewModel>(response);
 
+            var legalentities = await _mediator.SendAsync(new GetAccountLegalEntitiesByHashedAccountIdRequest
+            {
+                HashedAccountId = request.HashedAccountId
+            });
+
+            viewModel.LegalEntitiesCount = legalentities.LegalEntities.Count;
```

Comment:
With my performance hat on...this is a very inefficient way of getting the legal entity count.  Could you write a specific query to return just the count?

Reply:
Yes can do

Reply:
How about using the AccountStats query?

Reply:
Does too much unfortunately.

Reply:
Ok changes made